app/models/repository/destroying.rb

```diff
@@ -1,30 +1,58 @@
 module Repository::Destroying
   extend ActiveSupport::Concern
 
+  DELETION_WAIT_TIME = 30.days
+
   included do
-    @destroying ||= []
-    before_destroy :mark_as_destroying, prepend: true
-  end
+    scope :destroying, ->() { unscoped.where(is_destroying: true) }
+    scope :active, ->() { where(is_destroying: false) }
 
-  def is_destroying?
-    self.class.instance_variable_get(:@destroying).include?(self.id)
+    def self.find_deleted_repository_with_owner(path, user)
+      repository = Repository.find_by_path(path)
+      if user.owned_ids('Repository').include?(repository.id)
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.